contracts/Tokens/VTokens/VToken.sol

```diff
@@ -283,7 +283,21 @@ contract VToken is VTokenInterface, Exponential, TokenErrorReporter {
      * @return The supply interest rate per block, scaled by 1e18
      */
     function supplyRatePerBlock() external view returns (uint) {
-        return interestRateModel.getSupplyRate(getCashPrior(), totalBorrows, totalReserves, reserveFactorMantissa);
+        if (totalBorrows == 0) {
+            return 0;
+        }
+        uint256 utilizationRate = interestRateModel.utilizationRate(getCashPrior(), totalBorrows, totalReserves);
+        uint256 averageMarketBorrowRate = _averageMarketBorrowRate();
+        return
+            (averageMarketBorrowRate * utilizationRate * (mantissaOne - reserveFactorMantissa)) /
```

Comment:
here also we should use CarefulMath

Reply:
Fixed in commit[](https://github.com/VenusProtocol/venus-protocol/pull/210/commits/482b814841c2e68904e01250a89f48d5c745e83a)

Reply:
Fixed in [Commit](https://github.com/VenusProtocol/venus-protocol/pull/210/commits/ab515e83aaa817015e3c5515a17854a42388fa09)